app/_src/gateway/kong-enterprise/aws-iam-auth-to-rds-database.md

```diff
@@ -0,0 +1,121 @@
+
+{:.note}
+> **Note:** You must enable SSL connection on your Postgres DB instance, which is usually done by setting `ssl` parameter value to 1 in the parameter group. After enabling the IAM database authentication, you cannot disable the SSL connection.
+> **Note:** Currently, TLSv1.3 is not supported by RDS Postgres so you cannot set `pg_ssl_version` to `tlsv1_3`.
+
+- **Ensure your Kong instance have been assigned an IAM role**. If your Kong cluster are deployed without a proper IAM role, you must assign one first. For example, if your Kong instance is running on an EC2 machine, you can assign EC2 IAM role on your EC2 machine; if you're running Kong in the ECS cluster, you can assign a IAM role as a task role to your ECS service task definition; if you're running Kong in the EKS cluster, you can configure a kubernetes service account that can assume to your assigned role, and configure your pods to use the service account. The Kong instance can discover and fetch the AWS credential to use the assigned IAM role automatically, just like the AWS SDK. For now, it supports fetching through the following ways(in order of precedence):
+  - Use Environment variables like `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
+  - Use credential file provided by the environment variable `AWS_SHARED_CREDENTIALS_FILE`.
+  - Use [ECS Task IAM role](https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-iam-roles.html) automatically when the Kong instance is running inside an ECS container
+  - Use [IAM role defined by serviceaccount](https://docs.aws.amazon.com/eks/latest/userguide/associate-service-account-role.html) inside EKS pod
+  - Use [EC2 IAM role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html) inside EC2 environment
+
+{:.warning}
+> You **cannot** change the value of the environment variables that used to provide the AWS Credential after Kong instance boot up. The change will be ignored and **will not take effect** inside Kong instance.
+
+- **Ensure the IAM role you assinging to Kong has proper IAM policy**. Below is a simple example policy allows a user "john" to connect to a RDS instance "db-ABCDEFGHIJKL01234" by using the IAM database authentication:
+
+```json
+{
+   "Version": "2012-10-17",
+   "Statement": [
+      {
+         "Effect": "Allow",
+         "Action": [
+             "rds-db:connect"
+         ],
+         "Resource": [
+             "arn:aws:rds-db:us-east-2:1234567890:dbuser:db-ABCDEFGHIJKL01234/john"
+         ]
+      }
+   ]
+}
+```
+
+For more details, please check the AWS documentation: [https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.IAMPolicy.html](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.IAMPolicy.html)
+
+- **Ensure you have the database account created in the RDS**. You need to create the database user and then grant them the `rds_iam` role as shown in the following example:
```

Comment:
It would be of great help to first provide a code snippet of how to connect to RDS via `psql` and/or any other tool as a first step prior to creating the user and initial database.

```text
psql "host=db-your-instance-1.c4ka7ay7xeco.us-east-2.rds.amazonaws.com port=5432 user=testuser password=kongtest 
```

This will make the onboarding smoother.

Reply:
Done, good suggestion!